src/PostgREST/DbStructureTypes.hs

```diff
@@ -0,0 +1,227 @@
+{-# LANGUAGE DeriveAnyClass        #-}
+{-# LANGUAGE DeriveGeneric         #-}
+
+module PostgREST.DbStructureTypes where
```

Comment:
This is still a huge amount of types in here. It's the same as the `DbStructure.hs` file itself - which is a mere 841 lines right now. And the types are not together with the function that "naturally fit".

What about combining and splitting `DbStructure` and `DbStructureTypes` along the following lines:
- `PostgREST.Database.Tables`: "Tables" as in "Table-like" with views, mat view, ... (including Columns and PKs - and also view's "source columns")
- `PostgREST.Database.Relations`: FKs + everything that follows
- `PostgREST.Database.Functions`: what is currently called `Proc`, but that's technically not correct.
- `PostgREST.Database`: Basically just `getDbStructure` as a starter

I think the types from `Functions` should separate nicely from the others. `Relations` depends on some types from `Tables`, but that should be it.

Reply:
Note: `PostgREST.Database` could also hold the pg version stuff - but I'll probably get there in a later commit and then see what you have right now.

Reply:
Yes, that makes sense! In order to keep this mostly a 'shift things around' PR, I suggest we go with the following names and structure for now:

* `PostgREST.DbStructure` containst the top `DbStructure` type + the current loading code, as you suggest
* `PostgREST.DbStructure.Table` for `Table`, `Columns` with related types and functions
* `PostgREST.DbStructure.Relation` for `Table`, `Columns` with related types and functions
* `PostgREST.DbStructure.Proc` for the Proc type and related functions (to be renamed)
* `PostgREST.DbStructure.PgVersion`
* `PostgREST.ApiRequest.Preferences`

Renaming would then be a 2nd step. WDYT?

Reply:
Works for me!